server/src/main/java/org/elasticsearch/cluster/metadata/IndexTemplateV2.java

```diff
@@ -88,13 +95,15 @@ public static IndexTemplateV2 parse(XContentParser parser) throws IOException {
     }
 
     public IndexTemplateV2(List<String> indexPatterns, @Nullable Template template, @Nullable List<String> componentTemplates,
-                           @Nullable Long priority, @Nullable Long version, @Nullable Map<String, Object> metadata) {
+                           @Nullable Long priority, @Nullable Long version, @Nullable Map<String, Object> metadata,
+                           @Nullable DataStreamTemplate dataStreamTemplate) {
```

Comment:
Since this is part of the HLRC, I wonder if we should stay backwards compatible with the original constructor, i.e., have both variants? Not sure of the exact policy on that subject though.

Reply:
I think changing this constructor now is okay, because index templates v2 is experimental in 7.8? I don't think we have a real policy regarding hlrc bwc.

Reply:
I added the constructor:  https://github.com/elastic/elasticsearch/pull/55377/commits/5ed50fe948a24505257bfcae8d4226037a4e399a